### PR TITLE
Features/51383 custom content per profile attribute

### PR DESCRIPTION
**Pull request by neexite**

### CODE REVIEW

File: src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        protected ICustomerDefinitionsRepository CustomerDefinitionsRepository { get; }
+
+        public CustomerConditionProvider(IComposerContext composerContext,ICustomerRepository customerRepository, ICustomerDefinitionsRepository customerDefinitionsRepository)
+        {
+            ComposerContext = composerContext;
+            CustomerRepository = customerRepository ?? throw new ArgumentNullException(nameof(customerRepository));
+            CustomerDefinitionsRepository = customerDefinitionsRepository ?? throw new ArgumentNullException(nameof(customerDefinitionsRepository));
+        }
+
+        private Dictionary<string, List<string>> Fields => new Dictionary<string, List<string>>()
```
Comment (napernik) [inline src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs:28]: if this collection never changes it should be "private static readonly"

File: src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                })
+                //remove unsupported types
+                .Where(a => a.Field.Type != null)
+                .ToDictionary(a => a.Name, a => a.Field);
+        }
+
+        public Dictionary<string, Field> GetFields()
+        {
+
+            var definition = CustomerDefinitionsRepository.GetCustomerDefinitionAsync().Result;
```
Comment (napernik) [inline src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs:87]: Is there caching in place for customer definitions?

File: src/Orckestra.Composer.CustomerConditionProvider/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("ConditionalContent")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
```
Comment (napernik) [inline src/Orckestra.Composer.CustomerConditionProvider/Properties/AssemblyInfo.cs:11]: [assembly: AssemblyCompany("Orckestra A/S")]
[assembly: AssemblyCopyright("Copyright © Orckestra A/S 2022")]

File: src/Orckestra.Composer.CustomerConditionProvider/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("ConditionalContent")]
+[assembly: AssemblyDescription("")]
```
Comment (napernik) [inline src/Orckestra.Composer.CustomerConditionProvider/Properties/AssemblyInfo.cs:9]: "Adds customer related condition options for the Conditional Content feature."

File: src/Orckestra.Composer.CustomerConditionProvider/Repositories/CustomerDefinitionsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        protected ICacheProvider CacheProvider { get; }
+
+        public CustomerDefinitionsRepository(IOvertureClient overtureClient, ICacheProvider cacheProvider)
+        {
+            OvertureClient = overtureClient ?? throw new ArgumentNullException(nameof(overtureClient));
+            CacheProvider = cacheProvider ?? throw new ArgumentNullException(nameof(cacheProvider));
+        }
+        public virtual async Task<EntityDefinition> GetCustomerDefinitionAsync()
+        {
+            var definition = await OvertureClient.SendAsync(new GetCustomerDefinitionRequest()
```
Comment (napernik) [inline src/Orckestra.Composer.CustomerConditionProvider/Repositories/CustomerDefinitionsRepository.cs:22]: Please make sure the request is cached

File: src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs (hunk 1 above)
Comment (neexite) [reply src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs:28]: require C# 8.0

File: src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs (hunk 2 above)
Comment (neexite) [reply src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs:87]: in repo

**Pull request merged**